ehrql/dummy_data_nextgen/generator.py

```diff
@@ -160,21 +178,21 @@ def get_patient_data(self, patient_id, table_names):
         for name in table_names:
             # Seed the random generator per-table, so that we get the same data no
             # matter what order the tables are generated in
-            self.rnd.seed(f"{self.random_seed}:{patient_id}:{name}")
-            table_info = self.query_info.tables[name]
-            # Support specialised generators for individual tables, otherwise just make
-            # some empty rows
-            get_rows = getattr(self, f"rows_for_{table_info.name}", self.empty_rows)
-            rows = get_rows(table_info)
-            for row in rows:
-                # Fill in any values that haven't already been set by a specialised
-                # generator
-                self.populate_row(table_info, row)
-            table_node = table_info.table_node
-            column_names = table_node.schema.column_names
-            data[table_node] = [
-                (patient_id, *[row[c] for c in column_names]) for row in rows
-            ]
+            with self.seed(f"{patient_id}:{name}"):
+                table_info = self.query_info.tables[name]
+                # Support specialised generators for individual tables, otherwise just make
+                # some empty rows
+                get_rows = getattr(self, f"rows_for_{table_info.name}", self.empty_rows)
+                rows = get_rows(table_info)
+                for row in rows:
+                    # Fill in any values that haven't already been set by a specialised
+                    # generator
+                    self.populate_row(table_info, row)
```

Comment:
It might be clearer to just have the context manager wrap the section of code where it's used, which I think are just these lines here.